test/libcommonserver/io/testchecksetgetrights.cpp

```diff
@@ -0,0 +1,298 @@
+
+
+        IoError ioError = IoErrorUnknown;
+        CPPUNIT_ASSERT(IoHelper::createDirectory(path, ioError));
+        CPPUNIT_ASSERT(ioError == IoErrorSuccess);
+
+        CPPUNIT_ASSERT(IoHelper::createDirectory(subFolderPath, ioError));
+        CPPUNIT_ASSERT(ioError == IoErrorSuccess);
+
+        std::ofstream file(subFilePath.string());
+        CPPUNIT_ASSERT(file.is_open());
+        file << "testCheckSetAndGetRights";
+        file.close();
+
+
+        bool isReadable = false;
+        bool isWritable = false;
+        bool isExecutable = false;
+        bool exists = false;
+
+        bool result = IoHelper::setRights(path, true, true, true, ioError);
+        result = IoHelper::setRights(subFolderPath, true, true, true, ioError);
+        result = IoHelper::setRights(subFilePath, true, true, true, ioError);
+
+        CPPUNIT_ASSERT(IoHelper::getRights(path, isReadable, isWritable, isExecutable, exists));
+        CPPUNIT_ASSERT(exists && isReadable && isWritable && isExecutable);
+
+        CPPUNIT_ASSERT(IoHelper::getRights(subFolderPath, isReadable, isWritable, isExecutable, exists));
+        CPPUNIT_ASSERT(exists && isReadable && isWritable && isExecutable);
+
+        CPPUNIT_ASSERT(IoHelper::getRights(subFilePath, isReadable, isWritable, isExecutable, exists));
+        CPPUNIT_ASSERT(exists && isReadable && isWritable && isExecutable);
+
+        result = IoHelper::setRights(path, true, false, true, ioError);
+        result &= ioError == IoErrorSuccess;
+        if (!result) {
+            IoHelper::setRights(path, true, true, true, ioError);  // Restore the rights for delete
+            CPPUNIT_ASSERT(false /* Failed to set base rights */);
+        }
+
+        CPPUNIT_ASSERT(IoHelper::getRights(path, isReadable, isWritable, isExecutable, exists));
+        CPPUNIT_ASSERT(exists && isReadable && !isWritable && isExecutable);
+        CPPUNIT_ASSERT(IoHelper::getRights(subFolderPath, isReadable, isWritable, isExecutable, exists));
+        CPPUNIT_ASSERT(exists && isReadable && isWritable && isExecutable);
+        CPPUNIT_ASSERT(IoHelper::getRights(subFilePath, isReadable, isWritable, isExecutable, exists));
+        CPPUNIT_ASSERT(exists && isReadable && isWritable && isExecutable);
+
+        // Restore the rights
+        IoHelper::setRights(path, true, true, true, ioError);  // Restore the rights for delete
+    }
+
+    // Test on a non existing file
+    {
+        const SyncPath path = temporaryDirectory.path / "testCheckSetAndGetRights/nonExistingFile.txt";
+        bool isReadable = false;
+        bool isWritable = false;
+        bool isExecutable = false;
+        bool exists = false;
+        IoError ioError = IoErrorUnknown;
+
+        CPPUNIT_ASSERT(IoHelper::getRights(path, isReadable, isWritable, isExecutable, exists));
+        CPPUNIT_ASSERT(!exists);
+
+        CPPUNIT_ASSERT(IoHelper::setRights(path, true, true, true, ioError));
+        CPPUNIT_ASSERT(ioError == IoErrorNoSuchFileOrDirectory);
```

Comment:
```suggestion
        CPPUNIT_ASSERT_EQUAL(IoErrorNoSuchFileOrDirectory, ioError);
```

Reply:
I applied this suggestion in commit https://github.com/Infomaniak/desktop-kDrive/pull/68/commits/f12af122248f6dcb2d72c6ec08715e7c181b8fcd.